lib/controller-inputs/AnalogBusIn.cpp

```diff
@@ -0,0 +1,70 @@
+#include "AnalogBusIn.h"
+
+AnalogBusIn::AnalogBusIn(PinName p0, PinName p1, PinName p2, PinName p3, PinName p4, PinName p5, PinName p6,
+                         PinName p7) {
+  PinName pins[MAX_INPUTS_NUM] = {p0, p1, p2, p3, p4, p5, p6, p7};
+
+  // No lock needed in the constructor
+  _nc_mask = 0;
+  for (int i = 0; i < MAX_INPUTS_NUM; i++) {
+    _pin[i] = (pins[i] != NC) ? new AnalogIn(pins[i]) : 0;
+    if (pins[i] != NC) {
+      _nc_mask |= (1 << i);
+    }
+  }
+}
+
+AnalogBusIn::AnalogBusIn(PinName pins[MAX_INPUTS_NUM]) {
+  // No lock needed in the constructor
+  _nc_mask = 0;
+  for (int i = 0; i < MAX_INPUTS_NUM; i++) {
+    _pin[i] = (pins[i] != NC) ? new AnalogIn(pins[i]) : 0;
```

Comment:
I understand these concerns...so this is the same way mbed implements BusIn. I don't really want to use new if possible, but I think there is not a way to declare an analogin pin and then initialize it in the class constructor. I think this should be safe for our case since the AnalogBusIn are always created at the start of the app and we never declare any more of them during the app execution...given we don't dynamically allocate any additional things during the execution of the main body of the app. 
I'll make them smart pointers, I was just happily copying over code from the mbed lib lmao

Reply:
I see. To be fair it looks pretty safe. Here's how to use valgrind if u wanted to be extra careful. it's only a couple cli cmds:
https://stackoverflow.com/questions/5134891/how-do-i-use-valgrind-to-find-memory-leaks

Reply:
I don't think valgrind works in our case since the application is to be run on STM...
Since it is possible to do all the allocation at the start of the app in our case, we could simply print sth at the start of the app and see if it works

Reply:
u are right. this can be marked as resolved.